File: react/types/itineraryItemsRoute.ts
export type RouteWaypoint = {
  id: string;
  lat: number;
  lng: number;
  order: number;
};

export type ItineraryItemsRoute = {
  id: string;
  startItineraryItemId: string;
  endItineraryItemId: string;
  waypoints: RouteWaypoint[];
};

File: dotnet/Entites/ItineraryItemsRoute.cs
namespace dotnet.Entites
{
    public class ItineraryItemsRoute : BaseAuditableEntity
    {
        public Guid Id { get; set; }
        public Guid StartItineraryItemId { get; set; }
        public Guid EndItineraryItemId { get; set; }

        public ItineraryItem StartItineraryItem { get; set; }
        public ItineraryItem EndItineraryItem { get; set; }
        public ICollection<RouteWaypoint> Waypoints { get; set; } = new List<RouteWaypoint>(); // RouteWaypoints -> Later Update
    }
}
